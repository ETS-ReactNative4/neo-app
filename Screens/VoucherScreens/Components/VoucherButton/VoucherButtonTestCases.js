import React from "react";
import VoucherButton from "./VoucherButton";

const VoucherButtonTestCases = [
  {
    title: "Voucher Button",
    Component: <VoucherButton cta={"View Voucher"} action={() => null} />
  },
  {
    title: "Voucher Button Last",
    Component: (
      <VoucherButton cta={"View Voucher"} action={() => null} isLast={true} />
    )
  }
];

export default VoucherButtonTestCases;
