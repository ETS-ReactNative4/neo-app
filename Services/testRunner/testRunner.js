import renderer from "react-test-renderer";

const testRunner = testCase => {
  test(testCase.title, () => {
    const tree = renderer.create(testCase.Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export default testRunner;
