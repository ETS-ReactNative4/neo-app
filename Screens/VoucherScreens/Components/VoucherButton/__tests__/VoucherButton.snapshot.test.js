import VoucherButtonTestCases from "../VoucherButtonTestCases";
import testRunner from "../../../../../Services/testRunner/testRunner";

VoucherButtonTestCases.map(testRunner);
