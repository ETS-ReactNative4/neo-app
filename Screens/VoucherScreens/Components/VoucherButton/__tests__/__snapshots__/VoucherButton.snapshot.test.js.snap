// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Voucher Button 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "rgba(239, 239, 239, 1)",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingVertical": 24,
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgba(0,198,132,1)",
        "fontFamily": "SourceSansPro-SemiBold",
        "fontSize": 14,
        "lineHeight": 14,
      }
    }
  >
    View Voucher
  </Text>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "rgba(0,198,132,1)",
          "fontSize": 12,
        },
        undefined,
        Object {
          "fontFamily": "voyager",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Voucher Button Last 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingVertical": 24,
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgba(0,198,132,1)",
        "fontFamily": "SourceSansPro-SemiBold",
        "fontSize": 14,
        "lineHeight": 14,
      }
    }
  >
    View Voucher
  </Text>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "rgba(0,198,132,1)",
          "fontSize": 12,
        },
        undefined,
        Object {
          "fontFamily": "voyager",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;
